interpreter/ruby/ruby.go

```diff
@@ -847,10 +875,14 @@ func Loader(ebpf interpreter.EbpfHandler, info *interpreter.LoaderInfo) (interpr
 		// With Ruby 2.6 the field bp was added to rb_control_frame_t
 		// https://github.com/ruby/ruby/commit/ed935aa5be0e5e6b8d53c3e7d76a9ce395dfa18b
 		vms.control_frame_struct.size_of_control_frame_struct = 56
-	default:
+	case version < rubyVersion(3, 3, 0):
 		// 3.1 adds new jit_return field at the end.
 		// https://github.com/ruby/ruby/commit/9d8cc01b758f9385bd4c806f3daff9719e07faa0
 		vms.control_frame_struct.size_of_control_frame_struct = 64
+	default:
+		// 3.3+ size changed after re-ordering fields
+		// https://github.com/ruby/ruby/commit/7740526b1ccf62a027984e35375bb30ccbc0a000
```

Comment:
This seems like a wrong explanation; `#if VM_DEBUG_BP_CHECK` is enabled only for custom builds and most Ruby builds have the same layout before and after this change.

You're probably observing the removal of `bp` https://github.com/ruby/ruby/commit/f302e725e10ae05e613e2c24cae0741f65f2db91 that is mentioned on the `version < rubyVersion(3, 1, 0)` case.

Reply:
ah. I see. thanks. Yeah, was a bit lost trying to follow git history to add an explanation for this. That definitely makes more sense 😅